Default value is ignored for query parameters typed as lists

Environment:

- Python version: 3.12.3
- Operating system: ubuntu 22.04
- Flask version: 2.3.3
- flask-openapi3 version: 3.1.3

I defined a model with fields having default values like these ones:
```python
class ObjectQuery(BaseModel):
  order_by: list[str] = ["+creation", "+update"]
  status: list[ObjectStatus] = [ObjectStatus.NEW]
```
And use it as a query parameter in a route:
```python
@api.get("objects/", responses={HTTPStatus.OK: ObjectList})
def search_objects(query: ObjectQuery) -> Response:
```
The field correctly appear with their default values in the `openapi.json` file but when sending a query without any query parameter (as they all have default values) the instantiated `query` object fields are empty lists instead of the default values.

I checked the code and I found issue comes from [flask_openapi3.request._validate_query](https://github.com/luolingchun/flask-openapi3/blob/b073e03b5e4b5008b758175e9e4a8f4f35c44b3b/flask_openapi3/request.py#L36C1-L37C1). When the property is an array/list, the `value` set on line 41 is `[]`. So on line 44 the check `if value is not None:` is true and line 45 `query_dict[k] = value` set the value to `[]`. It leads to pydantic loading the value `[]` instead of its default value.

Is there an error in the way I define the default values?

Or is it a bug in flask_openapi3?
I guess it could check `if value is not None:` but maybe there is a valid usecase for clients to set explicitly an empty list or string or 0?
Or befure the `request_args.getlist()` a simple `request_args.get()` should be performed to check the value is present?

I tried to run the unit tests following the CONTRIBUTING.md but they fail with errors like
```
E   TypeError: __get_pydantic_core_schema__() takes 2 positional arguments but 3 were given
```
when loading test classes like this one:
```
class BookForm(BaseModel):
    file: FileStorage
    files: List[FileStorage]
    string: str
    string_list: List[str]
```

--- a/flask_openapi3/request.py
+++ b/flask_openapi3/request.py
@@ -41,7 +41,7 @@ def _validate_query(query: Type[BaseModel], func_kwargs):
             value = request_args.getlist(v.alias or k) or request_args.getlist(k)
         else:
             value = request_args.get(v.alias or k) or request_args.get(k)  # type:ignore
-        if value is not None:
+        if value is not None and value != []:
             query_dict[k] = value
     func_kwargs["query"] = query.model_validate(obj=query_dict)
 
@@ -69,7 +69,7 @@ def _validate_form(form: Type[BaseModel], func_kwargs):
                 value = json.loads(_value)  # type:ignore
             except (JSONDecodeError, TypeError):
                 value = _value  # type:ignore
-        if value is not None:
+        if value is not None and value != []:
             form_dict[k] = value
     func_kwargs["form"] = form.model_validate(obj=form_dict)
 

--- a/tests/test_list_with_default_value.py
+++ b/tests/test_list_with_default_value.py
@@ -1,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# @Author  : llc
+# @Time    : 2024/9/29 10:36
+from typing import List
+
+import pytest
+from pydantic import BaseModel
+
+from flask_openapi3 import OpenAPI
+
+app = OpenAPI(__name__)
+app.config["TESTING"] = True
+
+
+@pytest.fixture
+def client():
+    client = app.test_client()
+
+    return client
+
+
+class BookQuery(BaseModel):
+    age: List[int] = [1, 2]
+
+
+class BookForm(BaseModel):
+    age: List[float] = [3, 4]
+
+
+@app.get("/query")
+def api_query(query: BookQuery):
+    assert query.age == [1, 2]
+    return {"code": 0, "message": "ok"}
+
+
+@app.post("/form")
+def api_form(form: BookForm):
+    assert form.age == [3, 4]
+    return {"code": 0, "message": "ok"}
+
+
+def test_query(client):
+    client.get("/query")
+
+
+def test_form(client):
+    client.post("/form")

--- a/tests/test_request.py
+++ b/tests/test_request.py
@@ -106,7 +106,7 @@ def api_raw(raw: BookRaw):
 
 
 def test_query(client):
-    r = client.get("/query")
+    r = client.get("/query?age=1")
     print(r.json)
     assert r.status_code == 200
 

